synapse/handlers/relations.py

```diff
@@ -259,13 +268,9 @@ async def redact_events_related_to(
                     e.msg,
                 )
 
-    async def get_annotations_for_event(
-        self,
-        event_id: str,
-        room_id: str,
-        limit: int = 5,
-        ignored_users: FrozenSet[str] = frozenset(),
-    ) -> List[JsonDict]:
```

Comment:
The `room_id` is now check as part of the query (we check the parent and child event have the same room ID), but we lose checking the limit here.

Is that OK? Do we want to try to enforce it a different way?